common/utils/src/main/scala/org/apache/spark/internal/README.md

```diff
@@ -45,3 +45,52 @@ logger.error("Failed to abort the writer after failing to write map output.", e)
 ## Exceptions
 
 To ensure logs are compatible with Spark SQL and log analysis tools, avoid `Exception.printStackTrace()`. Use `logError`, `logWarning`, and `logInfo` methods from the `Logging` trait to log exceptions, maintaining structured and parsable logs.
+
+## External third-party ecosystem access
+
+* If you want to output logs in `scala code` through the structured log framework, you can define `custom LogKeys` as follows:
+
+```scala
+object CustomLogKeys {
```

Comment:
let's not mention CustomLogKeys here. It is not necessary, right?

Reply:
Okay